src/pages/Signup/Signup.js

```diff
@@ -0,0 +1,150 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+import { useAuth } from "../../context/AuthContext";
+import "./Signup.css";
+
+export const Signup = () => {
+  const [formData, setFormData] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+  });
+
+  const {
+    state: { error },
+    signupHandler,
+  } = useAuth();
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [isPasswordMatched, setIsPasswordMatched] = useState(true);
+  const { firstName, lastName, email, password } = formData;
+  const errorMsg = error !== "" ? error.data.errors[0] : "";
+
+  const submitSignupFormData = () => {
+    if (
+      firstName !== "" &&
+      lastName !== "" &&
+      email !== "" &&
+      password !== ""
+    ) {
+      signupHandler(firstName, lastName, email, password);
+      setFormData({
+        firstName: "",
+        lastName: "",
+        email: "",
+        password: "",
+      });
+      setConfirmPassword("");
+    }
+  };
+
+  return (
+    <main class="main-wrapper">
+      <section class="signup-section signup-container">
+        <div class="card signup-card-container">
+          <div class="card-title">Signup</div>
+          {errorMsg !== "" ? (
+            <div className="signup-error-message">{errorMsg}</div>
+          ) : null}
+          <div class="card-body">
+            <form onSubmit={e => e.preventDefault()}>
```

Comment:
Any specific reason to not  use the `onSubmit` event for handling form submit?

Reply:
No, i just want to put login handler in onClick of button.